01-messaging.md

```diff
@@ -476,6 +476,75 @@ every message maximally).
 Finally, the usage of periodic `ping` messages serves to promote frequent key
 rotations as specified within [BOLT #8](08-transport.md).
 
+### Peer storage
+
+## The `peer_storage` and `your_peer_storage` Messages
+
+Nodes that advertise the `option_provide_storage` feature offer storing
+arbitrary data for their peers. The data stored must not exceed 65531 bytes,
+which lets it fit in lightning messages.
+
+Nodes that advertise the `option_want_storage` feature want to have their
+data stored by their peers that support `option_provide_storage`.
+
+Nodes can verify that their `option_provide_storage` peers correctly store
+their data at each reconnection, by comparing the contents of the
+retrieved data with the last one they sent. However, nodes should not expect
+their peers to always have their latest data available.
+
+Nodes ask their peers to store data using the `peer_storage` message & expect peers to return them latest data using `your_peer_storage` message:
```

Comment:
nit: I'm not a fan of these message names. Can I suggest

`blob_store` instead of `peer_storage` and `blob_retrieval`. Perhaps it's not easy to come up with truly satisfactory names, but I'd love to avoid using "your" anywhere because of the level of context required to reason about who is on what side of the connection etc.

Reply:
Yes, these are good suggestions but the names should suggest that they come from one of your peers. blob_* doesn't imply that.

Reply:
Don't all of the messages come from our peers?

Reply:
I meant peer_storage suggests that the blob inside the msg is coming from your peer's storage, `blob` is a bit abstract.

Although I agree using `your` is not a good idea. How about `peer_storage_retrieval`?

Reply:
Sorry, I meant "blob_store" might not necessarily imply that the data comes from a peer's storage. Unlike "peer_storage," which explicitly denotes that the data is stored by peers within the network.

Although I agree using `your` is not a good idea. How about `peer_storage_retrieval`?

Reply:
Yeah and maybe we can label the other one `peer_storage_request`. Makes it pretty clear the roles each one plays.

Reply:
Yes, this is a good suggestion, I'll modify the doc. Thanks! :)